syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.imageviewer";
option java_outer_classname = "SendingImagesProto";
option objc_class_prefix = "HLW";

package imageviewer;

service SendingImageData {
    rpc sendImageData (ImageDataMessage) returns (ServerConfirmation) {}
}

service SendingMouseData {
    rpc sendMouseData (MouseMessage) returns (ServerConfirmation) {}
}

service SendingQuakeEvent {
    rpc sendQuakeEventData (QuakeEventMessage) returns (ServerConfirmation) {}
    rpc sendInjuryData (InjuryDataMessage) returns (ServerConfirmation) {}
}

message ImageDataMessage {
    bool negative = 1;
    bool neutral = 2;
    bool positive = 3;
    bool black_screen = 4;
    bytes image_name = 5;
}

message MouseMessage {
    int64 timestamp = 1;
    int32 move_x = 2;
    int32 move_y = 3;
}

message MouseButtonMessage {
    int64 timestamp = 1;
    int32 move_x = 2;
    int32 move_y = 3;
}

message InjuryDataMessage {
    int64 timestamp = 1;
    int32 injury_id = 2;
    int32 player_id = 3;
	int32 new_hp_value = 4;
}

message QuakeEventMessage {
    int64 timestamp = 1;
    int32 event_id = 2;
    int32 player_id = 3;
}

message ServerConfirmation {
    bytes confirm = 1;
}
